### PR TITLE
Refactored process.sparks getting ready to create mask line for when …

### PR DESCRIPTION
**Pull request by dblodgett-usgs**
…gages are offline

I'm going to do some more with sparks, but this should be a separate PR.

### CODE REVIEW

File: viz.yaml [hunk 1]
```diff
@@ -334,18 +334,9 @@ process:
...
-    scripts: scripts/process/process.sparks.R
-    reader: rds
-    depends: 
-      gage-data: storm-gage-height
-      sites: order-sites
-  - 
-    id: flood-sparks
-    location: cache/flood-sparks.rds
-    processor: flood_sparks
+    id: sparks
```
Comment (jread-usgs) [inline viz.yaml]: why aren't we displaying flood vs non flood anymore?

Comment (dblodgett-usgs) [reply viz.yaml]: We are, look down in the hurricane map visualize. I split it apart later.

File: scripts/process/process.sparks.R [hunk 2]
```diff
@@ -50,24 +25,32 @@ grab_clip_rect <- function(vals, flood){
...
-# Should rename at some point. Not always going to be discharge.
-process.flood_sparks <- function(viz = as.viz('flood-sparks')){
+process.sparks <- function(viz = as.viz('sparks')){
   library(dplyr)
   depends <- readDepends(viz)
   checkRequired(depends, c("gage-data", "sites", "nws-data"))
-  
   sites <- depends[["sites"]]
+  gage_data <- depends[["gage-data"]]$timestep_q
+  site_nils <- depends[["gage-data"]]$gage_mask_vals
```
Comment (dblodgett-usgs) [inline scripts/process/process.sparks.R]: This thing looks like: 
```
Browse[2]> str(site_nils)
 Named num [1:24] 4.75 4.92 1.53 2.36 11.2 ...
 - attr(*, "names")= chr [1:24] "50024950" "50027600" "50031200" "50035000" ...
```
So you can grab the nil value like: `mask_val <- site_nils[[sites$site_no[1]]]`

File: scripts/process/process.storm_gage_height.R [hunk 3]
```diff
@@ -11,17 +11,16 @@ process.storm_gage_height <- function(viz = as.viz("storm-gage-height")) {
   
   gage_height <- deps[["gage-height"]]
   
-  
-  gage_mins <- gage_height %>% 
+  gage_nils <- gage_height %>% 
     filter(p_Inst != no_value) %>%
     group_by(site_no) %>%
-    summarize(min = min(p_Inst))
+    summarize(nil = (min(p_Inst)-0.1)) # this -0.1 makes these mins unique but not mess with scaling.
```
Comment (dblodgett-usgs) [inline scripts/process/process.storm_gage_height.R]: This feels like a bit of a hack, but I think it'll work pretty well. Gives us a unique value to mask but doesn't mess with the plotting as it stands.

**Pull request merged**